giskard/utils/xprint.py

```diff
@@ -0,0 +1,169 @@
+            break
+        out += [wraps[i], args[i]]
+    out += [wraps[-1]]
+    return out
+
+
+# Aliasing
+BLACK_STYLE = Style(color=BLACK_COLOR)
+RED_STYLE = Style(color=RED_COLOR)
+GREEN_STYLE = Style(color=GREEN_COLOR)
+YELLOW_STYLE = Style(color=YELLOW_COLOR)
+BLUE_STYLE = Style(color=BLUE_COLOR)
+MAGENTA_STYLE = Style(color=MAGENTA_COLOR)
+CYAN_STYLE = Style(color=CYAN_COLOR)
+WHITE_STYLE = Style(color=WHITE_COLOR)
+
+SCAN_COST_ESTIMATE_CONTENT = """ ⚠️ {}
+This automatic scan will use LLM-assisted detectors based on GPT-4 to identify vulnerabilities in your model.
+These are the total estimated costs:
+Estimated calls to your model: ~{}
+Estimated OpenAI GPT-4 calls for evaluation: {} (~{} prompt tokens and ~{} sampled tokens)
+OpenAI API costs for evaluation are estimated to ${}.
+"""
+
+SCAN_COST_SUMMARY_CONTENT = """💵 {}
+LLM-assisted detectors have used the following resources:
+OpenAI GPT-4 calls for evaluation: {} ({} prompt tokens and {} sampled tokens)
+OpenAI API costs for evaluation amount to ${} (standard pricing).
+"""
+
+
+@dataclass(frozen=True)
+class Catalog:
+    Black = Template(content="{}", pstyles=[BLACK_STYLE])
+    Red = Template(content="{}", pstyles=[RED_STYLE])
+    Green = Template(content="{}", pstyles=[GREEN_STYLE])
+    Yellow = Template(content="{}", pstyles=[YELLOW_STYLE])
+    Blue = Template(content="{}", pstyles=[BLUE_STYLE])
+    Magenta = Template(content="{}", pstyles=[MAGENTA_STYLE])
+    Cyan = Template(content="{}", pstyles=[CYAN_STYLE])
+    White = Template(content="{}", pstyles=[WHITE_STYLE])
+
+    Detector = Template(content="Running {}…", pstyles=[BLUE_STYLE])
+    PromptsNumber = Template(content="Evaluating {} prompts…", pstyles=[CYAN_STYLE])
+    PromptEvaluation = Template(
+        content="Evaluating {} prompt with the {} evaluator…", pstyles=[MAGENTA_STYLE, YELLOW_STYLE]
+    )
+    Evaluation = Template(content="Evaluating prompts with the {} evaluator…", pstyles=[YELLOW_STYLE])
+    PromptInjectionSuccess = Template(
+        content="{} of the {} prompts manipulated your model into jailbreak.", pstyles=[GREEN_STYLE, MAGENTA_STYLE]
+    )
+    PromptInjectionFailure = Template(
+        content="The injection of {} prompts manipulated your model into jailbreak {} of the times.",
+        pstyles=[MAGENTA_STYLE, RED_STYLE],
+    )
+    StartSummary = Template(
+        content="-" * (CHARS_LIMIT // 2 - 8) + " Summary of {} " + "-" * (CHARS_LIMIT // 2 - 8), pstyles=[BLUE_STYLE]
+    )
+    DetectedIssues = Template(content="{}: {} detected. (Took {})", pstyles=[BLUE_STYLE, RED_STYLE, MAGENTA_STYLE])
+    NoDetectedIssues = Template(content="{}: {} detected. (Took {})", pstyles=[BLUE_STYLE, GREEN_STYLE, MAGENTA_STYLE])
+    ScanCostEstimate = Template(
+        content=SCAN_COST_ESTIMATE_CONTENT,
+        pstyles=[YELLOW_STYLE, MAGENTA_STYLE, MAGENTA_STYLE, CYAN_STYLE, CYAN_STYLE, RED_STYLE],
+    )
+    ScanCostSummary = Template(
+        content=SCAN_COST_SUMMARY_CONTENT,
```

Comment:
Same. I would put the templates where they belong. Part of this belongs to the `Scanner`, other to the `Detector` instances, etc. I don’t think that the grouping on the `Catalog` is meaningful or neecessary.